engine/sorted_collection/skiplist.hpp

```diff
@@ -67,21 +67,27 @@ struct SkiplistNode {
   // 4 bytes for alignment, the actually allocated size may > 4
   char cached_key[4];
 
-  static void DeleteNode(SkiplistNode* node) { free(node->heap_space_start()); }
+  static void DeleteNode(SkiplistNode* node, Allocator* alloc) {
```

Comment:
Since we have to call DeleteNode without a Skiplist instance, we may leave a TODO here. Some day we should have different SkiplistFactory to create Skiplist with nodes on DRAM/HBM/CXL mempool and each factory should destroy the skiplists created by it.

Reply:
The `alloc` argument is now removed from these 2 static methods: `NewNode()` and `DeleteNode()`. Will try to use an `allocator` in `thread_manager`, accessed by `thread local` data. Here, it also handles cases where the `thread_manager` is not set.